server/locks_controller.go

```diff
@@ -0,0 +1,143 @@
+package server
+
+import (
+	"bytes"
+	"fmt"
+	"html/template"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/events/locking"
+	"github.com/runatlantis/atlantis/server/events/models"
+	"github.com/runatlantis/atlantis/server/events/vcs"
+	"github.com/runatlantis/atlantis/server/logging"
+)
+
+// EventsController handles all webhook requests which signify 'events' in the
+// VCS host, ex. GitHub. It's split out from Server to make testing easier.
+type LockController struct {
+	AtlantisVersion    string
+	Locker             locking.Locker
+	Logger             *logging.SimpleLogger
+	VCSClient          vcs.ClientProxy
+	LockDetailTemplate TemplateWriter
+}
+
+var lockDeletedTemplate = template.Must(template.New("").Parse(
+	"The following Locks and plans deleted were deleted via the Atlantis UI:\n" +
+		"- path: `{{ .Path	}}` {{ .Workspace }}"))
+
+func (l *LockController) GetLockRoute(w http.ResponseWriter, r *http.Request) {
+	id, ok := mux.Vars(r)["id"]
+	if !ok {
+		l.respond(w, "No lock id in request")
+		return
+	}
+	idUnencoded, err := url.QueryUnescape(id)
+	if err != nil {
+		l.respond(w, "Invalid lock id", err)
+		return
+	}
+	lock, err := l.GetLock(idUnencoded)
+	t := l.GetLockTemplate(lock, id, idUnencoded)
+	if err != nil {
+		l.respond(w, "Invalid lock id", err)
+		return
+	}
+	err = l.LockDetailTemplate.Execute(w, t) // nolint: errcheck
```

Comment:
can remove the `nolint: errcheck` because you're actually checking the error now! :D